### PR TITLE
fix(trade): add amountsToSign to validation context and update valida…

### LINKED ISSUE #5726: Possible to place a Buy order with amount > balance
See the the thread: https://cowservices.slack.com/archives/C0361CDG8GP/p1747733810690919
2. Open Swap page, connect ti a wallet
4. Specify a By order
5. Fill in a but amount: 
- sell amount in the sell field should be < balance
- From (including costs) should be > balance
4. Check the Swap button

**AR**: the button is enabled. The order is placed, but it will never be executed

Image

**ER**: 'Insufficient balance' should be displayed when sell amount including costs and slippage > balance

### PR DESCRIPTION
**Pull request by fairlighteth**
# Summary

  Fixes #5726

  Fixed insufficient balance validation for buy orders where the validation was only checking
   against the input amount without including network costs and slippage. This allowed users
  to create orders that would fail execution due to insufficient balance when fees and
  slippage were factored in.

  # To Test

  1. **Buy order with sufficient balance (should work)**
     - [ ] Connect wallet with ~52 GHO balance on Arbitrum
     - [ ] Navigate to Swap page: GHO → USDC
     - [ ] Set buy amount to 51.3 USDC (this should show "From (incl. costs)" ~51.39 GHO)
     - [ ] Verify "Swap" button is enabled and order can be placed
     - [ ] Verify order executes successfully

  2. **Buy order with insufficient balance including slippage (should be blocked)**
     - [ ] With same ~52 GHO balance
     - [ ] Set buy amount to 51.5 USDC (this should show "From (incl. costs)" ~51.59 GHO)
     - [ ] Verify "Insufficient GHO balance" message appears
     - [ ] Verify button is disabled preventing order creation

  3. **Edge case: Balance exactly at slippage threshold**
     - [ ] Test amounts where balance is between "From (incl. costs)" and maximum slippage
  amount
     - [ ] Verify validation correctly blocks orders that would exceed balance with slippage

  4. **Sell orders and limit orders still work**
     - [ ] Verify sell orders (regular and limit) are unaffected by changes
     - [ ] Verify limit orders can still be placed with insufficient balance (existing
  behavior)

  # Background

  The issue was in the balance validation logic in `validateTradeForm.ts`. For buy orders,
  the system was only validating against `inputCurrencyAmount` (the base sell amount) rather
  than `maximumSendSellAmount` (which includes network fees and slippage protection).

  ## Example Scenarios

  **Before fix:**
  - User balance: 51.7444 GHO
  - Buy 51.5 USDC → shows "From (incl. costs): 51.59 GHO"
  - Validation only checked: 51.7444 > 51.585 ✅ (allowed)
  - Actual maximum spend with slippage: 51.85 GHO
  - Result: Order placed but fails execution (51.7444 < 51.85)

  **After fix:**
  - User balance: 51.7444 GHO
  - Buy 51.5 USDC → shows "From (incl. costs): 51.59 GHO"
  - Validation now checks: 51.7444 > 51.85 ❌ (blocked)
  - Result: "Insufficient GHO balance" - prevents failed orders

  The fix ensures users can only create orders they can actually execute, improving UX by
  preventing failed transactions while maintaining the existing behavior for limit orders
  that are designed to be placed with insufficient balance for future execution.

### CODE REVIEW

**Comment by fairlighteth** [comment]
Ready to review, but putting in DRAFT to not have 3 PRs open at a time.

File: apps/cowswap-frontend/src/modules/tradeFormValidation/services/validateTradeForm.ts [hunk 1]
```diff
@@ -8,43 +9,8 @@ import { ApprovalState } from 'common/hooks/useApproveState'
 
 import { TradeFormValidation, TradeFormValidationContext } from '../types'
 
-// TODO: Break down this large function into smaller functions
```
Comment (fairlighteth) [inline apps/cowswap-frontend/src/modules/tradeFormValidation/services/validateTradeForm.ts:11]: Went ahead, and did it.
